src/Microsoft.NetCore.Analyzers/Core/Runtime/DisposableFieldsShouldBeDisposed.cs

```diff
@@ -233,13 +241,13 @@ bool ShouldAnalyze(INamedTypeSymbol namedType)
                     // We only want to analyze types which are disposable (implement System.IDisposable directly or indirectly)
                     // and have at least one disposable field.
                     return !hasErrors &&
-                        namedType.IsDisposable(disposeAnalysisHelper.IDisposable) &&
+                        namedType.IsDisposable(disposeAnalysisHelper!.IDisposable) &&
```

Comment:
💭 It will sure be nice when the walker is updated to use call-site analysis like happens for definite assignment.